code-challenges.test.js

```diff
@@ -23,24 +30,136 @@ const people = [
+    it("iterates through array to return the remainders of the numbers only after division by 3 ", () => {
+      expect(timesThree(hodgepodge1)).toEqual([ 2, 0, -1, 0 ]);
+      expect(timesThree(hodgepodge2)).toEqual([ 2, 1, -1 ]);
+    });
+  });
+
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
+// PC: Input is an array containing numbers, strings and a boolean. Output will be an array containing only the numbers' remainder after being divided by 3. I know we will need to remove strings and booleans from the array, or isolate the numbers in a new array. Then use the modulo character to find and return the remainders after dividing by 3.
+// .filter iterates over the array and returns only the values with the typeof number
+// .map iterates over the smaller array and divides the numbers by 3 utilizing the modulo, returning the remainder as requested.
+
+const timesThree = (array) => {
+  return array.filter(value => 
+      typeof value === "number"
+      ) 
+   .map(value =>
+      value % 3
+      ) 
+}
+
+// PASS  ./code-challenges.test.js
+// peopleCaps
+//   ✓ Takes in an array and returns the values capitalized, and returned in a sentence (3 ms)
+// timesThree
+//   ✓ iterates through array to return the remainders of the numbers only after division by 3  (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.73 s
+// Ran all test suites.
+// ✨  Done in 2.46s.
+// learnacademy@learns-air week-6-assessment-VKathryn % 
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe("numCubed", () => {
+  it("takes in an array and returns the sum of the numbers cubed ", () => {
+    expect(numCubed(cubeAndSum1)).toEqual(99);
+    expect(numCubed(cubeAndSum2)).toEqual(1125);
+  });
+});
+
+// ReferenceError: numCubed is not defined GOOD FAIL
+
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 // b) Create the function that makes the test pass.
+
+// PC: Input is an array of numbers, output is the sum of all of the numbers cubed. Create a function that iterates over the array, cubes each of the numbers (n * n * n), adds the cubed numbers and returns the sum.
+// function numCubed takes in an array and uses .map to iterate through the array and cube each of the numbers using **3 
+// .reduce tells the function to iterate through the changed array and add each element to the one next to it, returning the sum of the cubed numbers. 
+
+const numCubed = (array) => {
+  return array.map(value =>
+      value **3
+      )
+      .reduce((a,b) => a + b, 0)
+}
+
```

Comment:
Such a good work flow. Appreciate that you place notes about the tools you are using. Very good.